src/query_graph/path_tree.rs

```diff
@@ -356,3 +356,165 @@ fn merge_conditions(
         (None, None) => None,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use apollo_compiler::{executable::DirectiveList, ExecutableDocument};
+    use petgraph::{stable_graph::NodeIndex, visit::EdgeRef};
+
+    use crate::{
+        error::FederationError,
+        query_graph::{
+            build_query_graph::build_query_graph,
+            condition_resolver::ConditionResolution,
+            graph_path::{OpGraphPath, OpGraphPathTrigger, OpPathElement},
+            path_tree::OpPathTree,
+            QueryGraph, QueryGraphEdgeTransition,
+        },
+        query_plan::operation::{
+            normalize_operation,
+            normalized_field_selection::{NormalizedField, NormalizedFieldData},
+        },
+        schema::{position::SchemaRootDefinitionKind, ValidFederationSchema},
+    };
+
+    // NB: stole from operation.rs
```

Comment:
whats NB?

Reply:
Yes. It was pretty widely used back at Synopsys. Maybe because it's shorter than "Note", I think.